front/src/store/modules/trip.js

```diff
@@ -0,0 +1,54 @@
+import axios from 'axios';
+import config from '@config';
+
+const { serverPath } = config;
+
+export default {
+  namespaced: true,
+  state: {
+    tripsHistory: []
+  },
+  mutations: {
+    INIT_TRIPS_HISTORY (state, trips) {
+      state.tripsHistory = trips;
+    },
+    REQUEST_TRIPS_HISTORY (state, trips) {
+      state.tripsHistory = trips;
+    },
+    FILTER_TRIPS_HISTORY (state, data) {
+      let filterFunc;
+
+      if (data.year && data.month) {
+        filterFunc = function (trip) {
+          return new Date(trip.startDate).getFullYear() === data.year
+            && new Date(trip.startDate).getMonth() + 1 === data.month;
+        };
+      } else if (data.year) {
+        filterFunc = function (trip) {
+          return new Date(trip.startDate).getFullYear() === data.year;
+        };
+      }
+
+      const trips = state.tripsHistory.filter(filterFunc);
+      state.tripsHistory = trips;
```

Comment:
it could be better to make via getters, you can think about it

Reply:
moved to the getters. I have named this getter 'getTripsHistoryByPeriod'